From 068c8b508a0501de9bda9d6d9a8c4ddc06cf7ff8 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 4 Nov 2020 11:25:02 +0000
Subject: [PATCH] overlays: Deprecate and delete the sdtweak overlay

The sdtweak overlay has been superseded by the board-specific
sd_* parameters such as sd_poll_once, sd_overclock etc.

For example, replace:

    dtoverlay=sdtweak,poll_once

with:

    dtparam=sd_poll_once

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/Makefile           |  1 -
 arch/arm/boot/dts/overlays/README             | 27 +++----------------
 arch/arm/boot/dts/overlays/overlay_map.dts    |  4 +++
 .../arm/boot/dts/overlays/sdtweak-overlay.dts | 25 -----------------
 4 files changed, 8 insertions(+), 49 deletions(-)
 delete mode 100644 arch/arm/boot/dts/overlays/sdtweak-overlay.dts

--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -158,7 +158,6 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	sc16is752-spi1.dtbo \
 	sdhost.dtbo \
 	sdio.dtbo \
-	sdtweak.dtbo \
 	sh1106-spi.dtbo \
 	smi.dtbo \
 	smi-dev.dtbo \
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2409,29 +2409,10 @@ Load:   <Deprecated>
 
 
 Name:   sdtweak
-Info:   Tunes the bcm2835-sdhost SD/MMC driver
-        N.B. This functionality is now available via the sd_* dtparams in the
-        base DTB.
-Load:   dtoverlay=sdtweak,<param>=<val>
-Params: overclock_50            Clock (in MHz) to use when the MMC framework
-                                requests 50MHz
-
-        force_pio               Disable DMA support (default off)
-
-        pio_limit               Number of blocks above which to use DMA
-                                (default 1)
-
-        debug                   Enable debug output (default off)
-
-        poll_once               Looks for a card once after booting. Useful
-                                for network booting scenarios to avoid the
-                                overhead of continuous polling. N.B. Using
-                                this option restricts the system to using a
-                                single card per boot (or none at all).
-                                (default off)
-
-        enable                  Set to off to completely disable the interface
-                                (default on)
+Info:   This overlay is now deprecated. Use the sd_* dtparams in the
+        base DTB, e.g. "dtoverlay=sdtweak,poll_once" becomes
+        "dtparam=sd_poll_once".
+Load:   <Deprecated>
 
 
 Name:   sh1106-spi
--- a/arch/arm/boot/dts/overlays/overlay_map.dts
+++ b/arch/arm/boot/dts/overlays/overlay_map.dts
@@ -61,6 +61,10 @@
 		deprecated = "use sdio,bus_width=1,gpios_22_25";
 	};
 
+	sdtweak {
+		deprecated = "use 'dtparam=sd_poll_once' etc.";
+	};
+
 	spi0-cs {
 		renamed = "spi0-2cs";
 	};
--- a/arch/arm/boot/dts/overlays/sdtweak-overlay.dts
+++ /dev/null
@@ -1,25 +0,0 @@
-/dts-v1/;
-/plugin/;
-
-/* Provide backwards compatible aliases for the old sdhost dtparams. */
-
-/{
-	compatible = "brcm,bcm2835";
-
-	fragment@0 {
-		target = <&sdhost>;
-		frag0: __overlay__ {
-			brcm,overclock-50 = <0>;
-			brcm,pio-limit = <1>;
-		};
-	};
-
-	__overrides__ {
-		overclock_50     = <&frag0>,"brcm,overclock-50:0";
-		force_pio        = <&frag0>,"brcm,force-pio?";
-		pio_limit        = <&frag0>,"brcm,pio-limit:0";
-		debug            = <&frag0>,"brcm,debug?";
-		enable           = <&frag0>,"status";
-		poll_once        = <&frag0>,"non-removable?";
-	};
-};
